libs/infinity_emb/infinity_emb/sync_engine.py

```diff
@@ -6,7 +6,16 @@
 import weakref
 from concurrent.futures import Future
 from functools import partial
-from typing import TYPE_CHECKING, Awaitable, Callable, Iterator, TypeVar
+from typing import (
+    TYPE_CHECKING,
+    Awaitable,
+    Callable,
+    Iterator,
+    TypeVar,
+    Union,
+    Sequence,
+)
+from PIL import Image
```

Comment:
PIL is not by default installed on all installations. This will break some people's code.

Potentially, use a different type hint here. Maybe in `primitives.py` add a 

```
from typing import TYPE_CHECKING, Any

PIL_IMAGE = Any
if TYPE_CHECKING:
     try:
           # fixes type annotation
           from PIL.IMAGE  import IMAGE  as PIL_IMAGE  
     except:
           pass

ImageTypeHint = PIL_IMAGE

Reply:
@michaelfeil have achieved this in the newest commit and please run the latest workflow. 
https://github.com/michaelfeil/infinity/actions/runs/10859326386

How do I close automatic sync of my fork workflow to your main repository? I just want to run them in my fork and select the success one lmao.

Reply:
Not sure why i still need to approve your workflow. Maybe because you have not commited to main yet.

Reply:
When I contribute somewhere, I adopt the following flow:
- fork the repo
- open a branch ("mf-new-feature-harness") on my forked repo (michaelfeil/lm-eval-harness)
- open a PR from michaelfeil/lm-eval-harness - mf-new-feature-harness to EleutherAI/lm-eval-harness - main. 
- sync the main branch from EleutherAi to my main as often as needed

As you opened the PR from your main to my main, this does not work as fast. But you could open new branches to try things out